stable/external-dns/templates/deployment.yaml

```diff
@@ -32,6 +32,16 @@ spec:
           {{- range $key, $value := .Values.extraArgs }}
             - --{{ $key }}={{ $value }}
           {{- end }}
+          env:
+          {{- if and .Values.cloudflare.api_key .Values.cloudflare.email }}
+          - name: CF_API_KEY
+            valueFrom:
+              secretKeyRef:
+                name: {{ template "external-dns.fullname" . }}
+                key: cloudflare_api_key
+          - name: CF_API_EMAIL
+            value: {{ .Values.cloudflare.email }}
```

Comment:
Wrap templates in quotes if they simply echo user-provided strings (e.g. `value: "{{ .Values.cloudflare.email }}"`).